docs/multi-stage-query/reference.md

```diff
@@ -424,6 +424,7 @@ The following table describes error codes you may encounter in the `multiStageQu
 | <a name="error_ColumnNameRestricted">`ColumnNameRestricted`</a> | The query uses a restricted column name. | `columnName`: The restricted column name. |
 | <a name="error_ColumnTypeNotSupported">`ColumnTypeNotSupported`</a> | The column type is not supported. This can be because:<br /> <br /><ul><li>Support for writing or reading from a particular column type is not supported.</li><li>The query attempted to use a column type that is not supported by the frame format. This occurs with ARRAY types, which are not yet implemented for frames.</li></ul> | `columnName`: The column name with an unsupported type.<br /> <br />`columnType`: The unknown column type. |
 | <a name="error_InsertCannotAllocateSegment">`InsertCannotAllocateSegment`</a> | The controller task could not allocate a new segment ID due to conflict with existing segments or pending segments. Common reasons for such conflicts:<br /> <br /><ul><li>Attempting to mix different granularities in the same intervals of the same datasource.</li><li>Prior ingestions that used non-extendable shard specs.</li></ul>| `dataSource`<br /> <br />`interval`: The interval for the attempted new segment allocation. |
+| <a name="error_InsertCannotAllocateSegment">`InsertAllocatedIncorrectSegment`</a> | The controller task could not allocate a new segment ID of the specified granularity due to conflict with existing segments or pending segments. <br /> <br /> This happens when a coarser segment is already overlapping the interval for which the allocation was requested. Either use a REPLACE to overwrite over the existing overlapping segment or re-run INSERT with the pre-existing segment granularity in order to append to the interval| `dataSource`<br /> <br />`requestedInterval`: The interval for the attempted new segment allocation. <br /><br />`allocatedInterval` The interval allocated for the requested segment |
```

Comment:
Nit: This fault says that the _requested_ segment could not be allocated, but something else could be.
In that sense, it doesn't seem too different from the `InsertCannotAllocateSegment`. 

Do you think just a different error message in the same fault type would work?